src/iexamine.cpp

```diff
@@ -1920,7 +1920,11 @@ void iexamine::plant_seed( player &p, const tripoint &examp, const itype_id &see
     }
     used_seed.front().set_age( 0_turns );
     g->m.add_item_or_charges( examp, used_seed.front() );
-    g->m.set( examp, t_dirt, f_plant_seed );
+    if( g->m.has_flag_furn( "PLANTABLE", examp ) ) {
+        g->m.furn_set( examp, furn_str_id( g->m.furn( examp ).obj().plant_transform ) );
```

Comment:
```suggestion
        g->m.furn_set( examp, g->m.furn( examp )->plant_transform ) );
```
Here and everywhere else: 
1. There's no need for `obj()`: it will be obsoleted and eventually removed.
2. Assuming `plant_transform` is already and instance of `furn_str_id`.